modules/planning/reference_line/reference_line.cc

```diff
@@ -626,18 +627,56 @@ bool ReferenceLine::GetSLBoundary(const common::math::Box2d& box,
   double end_l(std::numeric_limits<double>::lowest());
   std::vector<common::math::Vec2d> corners;
   box.GetAllCorners(&corners);
+
+  // the order must be counter-clockwise
+  std::vector<SLPoint> sl_corners;
   for (const auto& point : corners) {
     SLPoint sl_point;
     if (!XYToSL(point, &sl_point)) {
       AERROR << "failed to get projection for point: " << point.DebugString()
              << " on reference line.";
       return false;
     }
+    sl_corners.push_back(std::move(sl_point));
```

Comment:
Cause undefined behavior as the sl_point will be still in use later.

Reply:
No, it will not. The variable only exists in the for loop and will not be accessed after the line.

Reply:
My bad, copy-pasted some code after the line. Will fix.